### PR TITLE
[ENH] Add `avg_method` to `plot_surf_stat_map`

### LINKED ISSUE #4296: [BUG] `avg_method` and `symmetric_cbar` flags using `plot_surf_stat_map` working properly with plotly engine
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

Debian testing (trixie)

### Python version

- [ ] 3.12
- [X] 3.11
- [ ] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

0.10.3

### Expected behavior

When using the plotly engine, the colorbar should not be symmetric if symmetric_cbar=False. The avg_method should be user settable. This seems related to #3941 bug.

### Current behavior & error messages

This is what I got:

The flag symmetric_cbar=False is ignored, always plotting symmetric colorbars. On the other hand, when avg_method flag is inserted, I get the following error:

```bash
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 3
      1 from nilearn import plotting
----> 3 fig = plotting.plot_surf_stat_map(
      4     surf, texture, hemi='left',
      5     title='Surface left hemisphere', colorbar=True,
      6     engine='plotly', symmetric_cbar=False, avg_method='mean',
      7     darkness=0.5, cmap='black_red'
      8 )
     10 # change plotly lighting
     11 fig.figure.data[0].lighting.ambient = 0.5

File ~/.local/lib/python3.11/site-packages/nilearn/plotting/surf_plotting.py:1151, in plot_surf_stat_map(surf_mesh, stat_map, bg_map, hemi, view, engine, threshold, alpha, vmin, vmax, cmap, colorbar, symmetric_cbar, cbar_tick_format, bg_on_data, darkness, title, title_font_size, output_file, axes, figure, **kwargs)
   1142 # Call get_colorbar_and_data_ranges to derive symmetric vmin, vmax
   1143 # And colorbar limits depending on symmetric_cbar settings
   1144 cbar_vmin, cbar_vmax, vmin, vmax = get_colorbar_and_data_ranges(
   1145     loaded_stat_map,
   1146     vmin=vmin,
   1147     vmax=vmax,
   1148     symmetric_cbar=symmetric_cbar,
   1149 )
-> 1151 display = plot_surf(
   1152     surf_mesh, surf_map=loaded_stat_map,
   1153     bg_map=bg_map, hemi=hemi,
   1154     view=view, engine=engine, avg_method='mean', threshold=threshold,
   1155     cmap=cmap, symmetric_cmap=True, colorbar=colorbar,
   1156     cbar_tick_format=cbar_tick_format, alpha=alpha,
   1157     bg_on_data=bg_on_data, darkness=darkness,
   1158     vmax=vmax, vmin=vmin,
   1159     title=title, title_font_size=title_font_size, output_file=output_file,
   1160     axes=axes, figure=figure, cbar_vmin=cbar_vmin,
   1161     cbar_vmax=cbar_vmax, **kwargs
   1162 )
   1163 return display

TypeError: nilearn.plotting.surf_plotting.plot_surf() got multiple values for keyword argument 'avg_method'


```

### Steps and code to reproduce bug

```python

stat_img = image.smooth_img("TFCE_log_pDR_0002.nii", fwhm="fast")
surf = surface.load_surf_mesh('~/Downloads/Surf_Ice/BrainNet/BrainMesh_ICBM152_smoothed.lh.gii')
texture = surface.vol_to_surf(stat_img, surf, radius=3, kind='ball')

fig = plotting.plot_surf_stat_map(
    surf, texture, hemi='left',
    title='Surface left hemisphere', colorbar=True,
    engine='plotly', symmetric_cbar=False, avg_method='mean',
    darkness=0.5
)

# change plotly lighting
fig.figure.data[0].lighting.ambient = 0.5
fig.figure.data[0].lighting.diffuse = 0.6
fig.figure.data[0].lighting.specular = 0.1
fig.figure.data[0].lighting.roughness = 0.03
fig.figure.data[0].lighting.fresnel = 0.05
fig.figure.data[0].lightposition = dict(x=0, y=20, z=30)

fig.show()

```

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Related to #4296 

Changes proposed in this pull request:

- add `avg_method` parameter to `plot_surf_stat_map`
- throw warnings when using plotly and passing an unimplemented parameter to one of the surface plotting function to avoid users being confused why some parameters have no effect.
  - this required setting the default of a few parameters to `None` in the function declaration and using the original default when using the matplotlib engine.
  - this should now be reflected in the doc strings

### CODE REVIEW

File: nilearn/plotting/js_plotting_utils.py [hunk 1]
```diff
@@ -76,8 +76,8 @@ def colorscale(cmap, values, threshold=None, symmetric_cmap=True,
         warnings.warn('you have specified symmetric_cmap=False '
                       'but the map contains negative values; '
                       'setting symmetric_cmap to True',
-                      stacklevel=3)
-        symmetric_cmap = True
+                      stacklevel=4)
+        # symmetric_cmap = True
```
Comment (Remi-Gau) [inline nilearn/plotting/js_plotting_utils.py:80]: no `symmetric_cbar` handling in the code for plotly so trying to rely on `symmetric_cmap`

silencing the part that forces a `symmetric_cmap=true` just for testing

**Comment by Remi-Gau** [comment]
This PR would allow you to get something like this:

newplot


But this feels sub-optimal as what the original issue complains about is a using the assymetric color bar and not an assymetric color map

**Comment by Remi-Gau** [comment]
@bthirion 
Am not sure we can consider this PR as bug fix. What do you think?
If not a bug fix, am tempted to say we wait to merge this.

File: nilearn/plotting/surf_plotting.py [hunk 2]
```diff
@@ -1084,8 +1109,9 @@ def plot_surf_stat_map(surf_mesh, stat_map, bg_map=None,
 
         Default=True.
 
-    alpha : float or 'auto', default='auto'
+    alpha : float or 'auto' or None, default=None
         Alpha level of the :term:`mesh` (not the stat_map).
+        Will default to ``"auto"`` of ``None`` is passed.
```
Comment (bthirion) [inline nilearn/plotting/surf_plotting.py:1114]: So, why not keeping `auto` as  a default ?

Comment (bthirion) [inline nilearn/plotting/surf_plotting.py:1114]:
```suggestion
        Will default to ``"auto"`` if ``None`` is passed.
```

File: nilearn/plotting/surf_plotting.py [hunk 3]
```diff
@@ -1588,10 +1629,11 @@ def plot_surf_roi(surf_mesh,
 
         .. versionadded:: 0.7.1
 
-    alpha : float or 'auto', default='auto'
-        Alpha level of the :term:`mesh` (not the stat_map).
-        If default, alpha will default to 0.5 when no bg_map is passed
-        and to 1 if a bg_map is passed.
+    alpha : float or 'auto' or None, default=None
+        Alpha level of the :term:`mesh` (not surf_data).
+        Will default to ``"auto"`` of ``None`` is passed.
```
Comment (bthirion) [inline nilearn/plotting/surf_plotting.py:1634]:
```suggestion
        Will default to ``"auto"`` if ``None`` is passed.
``` + same comment as above

**Review by bthirion — COMMENTED** [review]
LGTM overall, thx !

**Review by bthirion — APPROVED** [review]
The change LGTM, thx.

**Pull request merged**